datahub-frontend/app/auth/sso/oidc/OidcAuthorizationGenerator.java

```diff
@@ -46,5 +55,32 @@ public CommonProfile generate(WebContext context, CommonProfile profile) {
         
         return profile;
     }
+
+    private static JWT parse(final String s) throws ParseException {
+        final int firstDotPos = s.indexOf(".");
+
+        if (firstDotPos == -1) {
```

Comment:
Do we need to write this logic ourselves? Cannot we ask the JWT (or JWS) Parser to parse, and then verify that the signing key exists and is in a supported set?

Reply:
It could throw an exception based on the class returned, however it is likely to result in a false positive when run against automated scans.